### PR TITLE
Allow specifying backend settings using an environment variable

### PR DESCRIPTION
**Pull request by BenjaminVadant**
A possible solution to https://github.com/xbill82/vue-kuzzle/issues/1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,30 +1,20 @@
...
-
-  if (process.env.NODE_ENV === 'production') {
-    // Define here which backend is to be used by the production build
-  }
-
-  const backend = backends[backendName] ? backends[backendName] : null;
-
-  if (!backend) {
-    throw new Error(`Unable to find backend ${backendName}`);
+  if (process.env.VUE_KUZZLE_BACKEND) {
```
Comment (xbill82) [inline index.js:16]: Let's take this whole check away.

File: index.js [hunk 2]
```diff
@@ -1,30 +1,20 @@
...
-      local: {
-        host: 'localhost',
-        options: {
-          port: 7512,
-          sslConnection: false
-        }
+const instantiateKuzzleSDK = (backendOptions) => {
+  let backend;
+
+  if (!backendOptions) {
```
Comment (xbill82) [inline index.js:6]: Let's just define the default backend options here:

```javascript
import { defaultsDeep } from 'lodash';

// ...

const backend = defaultsDeep(backendOptions, {
  host: 'localhost',
  options: {
    sslConnection: false
    port: 7512,
   }
 });
```

**Review by xbill82 — CHANGES_REQUESTED** [review]
I totally agree with the need of keeping only the concerned backend exposed in the built code, but I don't like the idea of passing a stringified object in an environment variable. By the way, I think that it's not this plugin's concern to figure out which backend config the user wants. I prefer to leave the management of the backend config at App level and make the plugin talke a `backendOptions` object. 
The plugin just know the defaults values for this object.

**Comment by scottinet** [comment]
This PR has been inactive for a long time => closing it.

**Pull request closed without merging**